geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionTest.java

```diff
@@ -276,4 +278,26 @@ public void doNormalMessageShouldNotProcessMessageWhenTerminated() {
     assertThat(spy.getProcessMessages()).isFalse();
     verify(spy, never()).resumeThreadMonitoring();
   }
+
+  @Test
+  public void cleanClientAuthShouldNullIt() {
+    ClientUserAuths clientUserAuths = mock(ClientUserAuths.class);
+    ServerConnection spy = spy(serverConnection);
+    spy.setClientUserAuths(clientUserAuths);
+    spy.cleanClientAuths();
+    assertThat(spy.getClientUserAuths()).isNull();
+  }
+
+  @Test
+  public void handleTerminationShouldNotNullClientAuths() {
+    when(acceptor.getClientHealthMonitor()).thenReturn(clientHealthMonitor);
```

Comment:
This test verifies only an implementation detail. If `handleTermination(...)` leaves `clientUserAuths` non-null, what effect does that have on the subsequent behavior of methods that a production caller can call? To test an actual responsibility, make the test call those methods and verify the results.

Also, under some conditions, `handleTermination` definitely nulls `clientUserAuths`, so there should be separate tests for the conditions where it nulls `clientUserAuths` and the conditions where it does not.

Reply:
added the other test as suggested to test the actual usage of `handleTermination` will null `clientUserAuths`

Reply:
These tests are better, but they still test internal implementation details. Instead of asserting the value of this field, is there a way to observe the _effects_ of the field being null or not null? That is, what later call does the right thing only when this field has the correct value?